File: Java/2399. Check Distances Between Same Letters.java
class Solution {
    public boolean checkDistances(String s, int[] distance) {
        int temp[] = new int[26], t;

        for(int i = 0; i < s.length(); i++){
            t = (int)s.charAt(i) - 97;
            if(temp[t] == 0) temp[t] = i + 1;
            else if(i - temp[t] != distance[t]) return false;
        }

        return true;
    }
}

File: Python/2399. Check Distances Between Same Letters.py
class Solution:
    def checkDistances(self, s: str, distance: List[int]) -> bool:
        temp = {}

        for i, x in enumerate(s):
            if x not in temp: temp[x] = i
            elif i - temp[x] - 1 != distance[ord(x) - 97]: return False
        
        return True
